core/templates/dev/head/pages/exploration_player/SolutionManagerService.js

```diff
@@ -27,6 +28,15 @@ oppia.factory('SolutionManagerService', [
       viewSolution: function () {
         if (!solutionHasBeenViewed) {
           solutionHasBeenViewed = true;
+
+          // Check that the current card is an active card by checking whether
+          // the active card's solution is the same as the current solution.
+          var activeStateName = playerPositionService.getCurrentStateName();
+          var activeCardSolution = $injector.get(
```

Comment:
Try not to use $injector. It bypasses the usual dependency management system and makes things harder to maintain. It is also a "code smell" that indicates that something hacky is going on.

I also don't think this is the right place to do this. The solution manager manages solutions. Let the player service handle event recording. E.g. if the solution button is clicked, tutor card calls a method in player service which then records the event.

Reply:
@seanlip I used $injector because there was a circular dependency issue. But, I'll look into changing the event recording to the player services.

Reply:
Yep, circular dependencies are also "code smells" indicating that something is a bit off about the design. It's basically the same issue, so in general, if you find that you have to use $injector, something is probably a bit wrong somewhere.

OK, sounds good. Thanks!

Reply:
Done.